app/models/orders_package.rb

```diff
@@ -76,10 +77,11 @@ def self.add_partially_designated_item(package)
   end
 
   private
-  def recalculate_quantity
+  def recalculate_quantity(operation)
     update_designation_of_package
     package = Package.find_by_id(package_id)
     package.update_in_stock_quantity(get_total_quantity)
+    StockitSyncOrdersPackageJob.perform_now(package.id, self, operation)
```

Comment:
Can't pass self here - must be self.id and the corresponding job must look up the OrderPackage from the database as this action is asynchronous and will happen sometime in the future.